text/0000-futures.md

```diff
@@ -808,4 +905,4 @@ model.
 # Unresolved questions
 [unresolved]: #unresolved-questions
 
-None at present
+- Final name for `FutureResult`.
```

Comment:
Names suggested so far:
- `FutureResult`
  - Pro: Word Future is in front, same word oder as `Iter` and `IterMut`
- `ResultFuture`
  - Pro: The word `Future` at the end emphasizes that every `ResultFuture` is a `Future`
- `FallibleFuture`
  - Pro: The word `Future` at the end emphasizes that every `ResultFuture` is a `Future`
  - Pro: Alliteration
  - Con: Not as descriptive: No indication that the `Result` type is involved

Reply:
Another option: `TryFuture`. In the same way that `TryFrom` is the fallible case of `From`, and the `Try` trait is the generalization of "a thing that might fail."